app/src/main/java/io/intrepid/contest/screens/entrysubmission/join/JoinContract.java

```diff
@@ -2,18 +2,22 @@
 
 import io.intrepid.contest.base.BaseContract;
 
-public class JoinContract {
+class JoinContract {
     interface View extends BaseContract.View {
         void showSubmitButton();
 
         void hideSubmitButton();
 
         void showEntryNameScreen();
+
+        void showInvalidCodeErrorMessage();
```

Comment:
Good additions. This could also be simplified so that you give the View interface one showErrorMessage that takes in as parameter the Error Msg or type of msg.